apps/ejabberd/src/mod_caps.erl

```diff
@@ -575,22 +575,22 @@ concat_info(Els) ->
                              Els)).
 
 concat_xdata_fields(Fields) ->
-    [Form, Res] = lists:foldl(fun(#xmlel{name = <<"field">>, attrs = Attrs, children = Els},
-                                  [FormType, VarFields] = Acc) ->
-                                      case xml:get_attr_s(<<"var">>, Attrs) of
-                                          <<"">> -> Acc;
-                                          <<"FORM_TYPE">> ->
-                                              [xml:get_cdata([xml:get_subtag(VarFields,
-                                                                             <<"value">>)]),
-                                               VarFields];
-                                          Var ->
-                                              [FormType,
-                                               [[[Var, $<], extract_values_sorted_cdatas(Els)]
-                                                | VarFields]]
-                                      end;
-                                 (_, Acc) -> Acc
-                              end,
-                              [<<"">>, []], Fields),
+    [Form, Res] =
+    lists:foldl(fun(#xmlel{name = <<"field">>, attrs = Attrs, children = Els} = FieldEl,
+                    [FormType, VarFields] = Acc) ->
+                        case xml:get_attr_s(<<"var">>, Attrs) of
```

Comment:
I see you used `exml_query:path` instead of `xml:get_cdata(...)`, that's good, thanks! Could you also change the `xml:gett_attr_s` with equivalent from `exml_query`?